pkg/catalog/fetch_node_details_vcluster.go

```diff
@@ -47,6 +47,9 @@ func (nodeDetails *NodeDetails) parseVNodeDetails(vnodeDetails *vclusterops.Node
 	nodeDetails.ReadOnly = vnodeDetails.IsReadOnly
 	nodeDetails.SandboxName = vnodeDetails.SandboxName
 	nodeDetails.ShardSubscriptions = int(vnodeDetails.NumberShardSubscriptions)
+	if nodeDetails.ShardSubscriptions > 0 {
```

Comment:
I suggest we leave a comment so that we know why we are decrementing the subscription count.
```suggestion
	// The shard subscriptions we get from vcluster includes the replica shard.
	// We decrement that by one to account for that. We want to know when there
	// are 0 shard subscriptions in order to drive a shard rebalance.
	if nodeDetails.ShardSubscriptions > 0 {
```